src/navigation/navigation_main.cc

```diff
@@ -117,31 +194,26 @@ void LaserCallback(const sensor_msgs::LaserScan& msg) {
     }
   }
   CHECK_EQ(cached_rays_.size(), msg.ranges.size());
-  static vector<Vector2f> point_cloud_;
   point_cloud_.resize(cached_rays_.size());
   for (size_t i = 0; i < cached_num_rays_; ++i) {
     const float r =
       ((msg.ranges[i] > msg.range_min && msg.ranges[i] < msg.range_max) ?
       msg.ranges[i] : msg.range_max);
     point_cloud_[i] = r * cached_rays_[i] + kLaserLoc;
   }
-  navigation_.ObservePointCloud(point_cloud_, msg.header.stamp.toSec());
-  last_laser_msg_ = msg;
+  return point_cloud_;
```

Comment:
This is setting and returning a global - to avoid confusion it should do one or the other: either set a local var and return that, or set the global var and have no return.